README.md

```diff
@@ -1,189 +1,337 @@
+`ispace(<type>, N)`, where N is the size of the array, and `<type>` is either
+int1d/int2d/int3d depending on the dimension of the transform. The fieldspace of
+the region is the type supported by the transform - e.g, in a real-to-complex
+transform with double precision, the input array will have fieldspace `double`
+and output array will have fieldspace `complex64`. Larger fieldspaces that
+contain the appropriate types can also be passed in via field polymorphism -
+[here](https://groups.google.com/g/legionusers/c/yvQa8BE6QD0/m/_1cL_w-aAAAJ) is
+an example.
+
+For example, in a 1D double-to-complex64 transform of size 3, the input and
+output regions may be initialized as follows:
+
+```lua
+var r = region(ispace(int1d, 3), double)
+var s = region(ispace(int1d, 3), complex64)
 ```
+
+The plan region always takes the following form, with fieldspace `fft.plan` (see
+`fft.rg` for description of plan fieldspace):
+
+```lua
 var p = region(ispace(int1d, 1), fft1d.plan)
-fft1d.make_plan(r, s, p)
 ```
 
-**Important:** `make_plan` overwites the input and output regions `r`
-and `s`. This is mandated by FFTW, which Regent FFT uses on CPUs. In
-order to avoid overwriting data, the user must either initialize the
-plan prior to loading the regions with data, or else must create a
-temporary region (of the same size and layout as the real one) for use
-in initialization.
+`make_plan` is a `__demand(__inline)` task. This means that if the user wants it
+to execute it in a separate task, they must wrap the task themselves.
 
-Note that, like all non-distributed mode tasks, `make_plan` is a
-`__demand(__inline)` task. This means that if the user wants it to
-execute it in a separate task, they must wrap the task themselves.
+> [!IMPORTANT]
+>
+> `make_plan` overwrites the input and output regions `r` and `s`. This is
+> mandated by FFTW, which Regent FFT uses on CPUs. In order to avoid overwriting
+> data, the user must either initialize the plan prior to loading the regions
+> with data, or else must create a temporary region (of the same size and layout
+> as the real one) for use in initialization.
 
-To initialize in distributed mode:
+#### 3. Execute Plan
 
+Next, we execute the plan. This takes the same 3 regions as mentioned above.
+
+```lua
+fft1d.execute_plan_task(r, s, p)
 ```
-var n = fft1d.get_num_nodes()
-var p = region(ispace(int1d, n), fft1d.plan)
-var p_part = partition(equal, p, ispace(int1d, n))
-fft1d.make_plan_distrib(r, r_part, s, s_part, p, p_part)
-```
 
-Note the use of `get_num_nodes` to determine the size of the `p`
-region and partition. The task `make_plan_distrib` is a
-`__demand(__inline)` task that internally performs an index launch
-over the machine to initialize `p`.
+Note that `execute_plan` is a `__demand(__inline)` task (similar to `make_plan`
+above). The task `execute_plan_task` is simply a wrapper around `execute_plan`
+for convenience, to avoid needing to define this explicitly.
```

Comment:
Could remove now if you take my suggestion above.

Reply:
Removed